src/KafkaFlow/Event.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace KafkaFlow
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    internal class Event<TArg> : IEvent<TArg>
+    {
+        private readonly ILogHandler logHandler;
+
+        private readonly IList<Func<TArg, Task>> handlers = new List<Func<TArg, Task>>();
```

Comment:
I've added the unsubscribe method

Reply:
I believe this implementation won't work because of the same thing you said in this comment https://github.com/Farfetch/kafkaflow/pull/438#discussion_r1342502489

Meaning that if you have:
```csharp
middlewareContext.Worker.WorkerStopped.Subscribe(() => this.TriggerDispatchAndWaitAsync());

// Then
middlewareContext.Worker.WorkerStopped.Unsubscribe(() => this.TriggerDispatchAndWaitAsync());
```

The `Unsubscribe` call won't clear the subscription as expected.

This will only work if:

```csharp
var func = () => this.TriggerDispatchAndWaitAsync();

middlewareContext.Worker.WorkerStopped.Subscribe(func);

// Then
middlewareContext.Worker.WorkerStopped.Unsubscribe(func);
```

This is very error-prone for the user and can easily become a memory leak in their system.

Reply:
Yup, you are right.

I've changed the implementation, the subscribe method returns an reference to the subscription (IEventSubscription) and it can be used to cancel the subscription.